tests/unit/test_charm.py

```diff
@@ -1,25 +1,147 @@
 #!/usr/bin/env python3
 
+"""Test default charm events such as upgrade charm, install, etc."""
+
 import unittest
+from unittest.mock import patch
 
 import ops.testing
-from ops.model import ActiveStatus
+from charm import SlurmdbdCharm
+from ops.model import ActiveStatus, BlockedStatus
 from ops.testing import Harness
 
-from charm import SlurmdbdCharm
+ops.testing.SIMULATE_CAN_CONNECT = True
 
 
 class TestCharm(unittest.TestCase):
-    def setUp(self):
-        ops.testing.SIMULATE_CAN_CONNECT = True
-        self.addCleanup(setattr, ops.testing, "SIMULATE_CAN_CONNECT", False)
-
+    def setUp(self) -> None:
         self.harness = Harness(SlurmdbdCharm)
         self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+
+    def test_install_success(self) -> None:
+        """Test that slurmdbd can successfully be installed."""
+        self.harness.charm.on.install.emit()
+        self.assertEqual(
+            self.harness.charm.unit.status, ActiveStatus("slurmdbd successfully installed")
+        )
+
+    @patch("slurm_ops_manager.SlurmManager.install")
+    def test_install_fail(self, install) -> None:
+        """Test that slurmdbd install fail handler works."""
+        install.side_effect = False
+        self.harness.charm.on.install.emit()
+        self.assertEqual(
+            self.harness.charm.unit.status, BlockedStatus("Error installing slurmdbd")
+        )
+
+    @patch("pathlib.Path.read_text", return_value="v1.0.0")
+    def test_upgrade_charm(self, _) -> None:
+        """Test that charm upgrade procedure works."""
+        self.harness.charm.on.upgrade_charm.emit()
+
+    @patch("ops.model.Unit.is_leader", return_value=True)
+    def test_is_leader(self, _) -> None:
+        """Test that _is_leader method works.
+
+        Notes:
+            The _is_leader method should be removed from charm.py
+            since _is_leader is already defined in ops.
+        """
+        self.assertEqual(self.harness.charm._is_leader(), True)
```

Comment:
I think it's more clear to change this to `self.assertTrue(self.harness.charm._is_leader())`. Same thing with other cases below to `self.assertFalse` instead of `self.assertEqual(..., False)`.

Reply:
Resolved in https://github.com/canonical/slurmdbd-operator/pull/2/commits/6ad37cd848f9a2dcbee3fb1df15b742d19f45260. I kept assertEqual but compare the returned value of the locally defined `_is_leader()` method with the returned value of ops `is_leader()` method.